### DISCUSSION TITLE
Desenha rota em Javascript

### DISCUSSION THREAD

**Discussion by quinhosp** [frontendbr/forum#2294] [Perguntas e Respostas]
boa tarde, sou iniciante em javascript.

Preciso escrever um script que leia um arquivo .gpx ou .txt e desenhe no navegador a rota.
Alguém pode me ajudar?

**Comment by igorlvicente** [comment]
O que você já tem?

**Comment by ghost** [comment]
Olá, sendo direta: não parece ser tão simples.

Tem 3 etapas e creio que no mínimo precisaria de um backend também, a não ser que já exista um serviço já pronto para contratar.

Você precisaria:
- Estudar alguma API de mapas (geralmente são pagas), e podem ou não fornecer roteirização já pra ti;
- Como você vai subir esses arquivos e pra onde;
- E quem leria esses arquivos e devolveria pra ti os dados para vc integrar na API de mapas.
